### ISSUE TITLE
Extra newlines appear in rendered README 

### ISSUE THREAD

**Issue by sunjay** [rust-lang/crates.io#965]
I'm so excited that README files now show up in crates.io! I started using it without even realising and then when I saw that it was there I was so happy! :smile: 

I noticed that the `css-color-parser` has a lot of extra newlines for some reason when it is shown on crates.io vs. when it is shown on GitHub. There is a lot of extra space between bullets and after paragraphs which GitHub is able to collapse but crates.io does not.

Looking at the README, the author of that package seems to have added a `<br>` after every line for whatever reason. This is further exacerbated on crates.io in certain places because the inserted paragraph tags have additional space due to the `<br>`s added everywhere (see just after the word Reliable in the README). Crates.io should probably collapse all of that extra space together in the same way that GitHub does.

GitHub: https://github.com/7thSigil/css-color-parser-rs
Crates.io: https://crates.io/crates/css-color-parser

## Screenshots: (click to view full size)

![screencapture-crates-io-crates-css-color-parser-1503115712895](https://user-images.githubusercontent.com/530939/29483379-2e899088-8462-11e7-84e1-7eb748845b22.png)

![screencapture-github-7thsigil-css-color-parser-rs-1503115691837](https://user-images.githubusercontent.com/530939/29483378-2c9e4ebc-8462-11e7-867f-aad30d4ebdad.png)

**Comment by anthonynguyen** [comment]
Seems like the Markdown renderer renders this just fine, but after Ammonia sanitizes it, all the `<br>s` get closed and that's what does the weird stuff.

----

##### original.md

```markdown
Easy-to-use Rust parser for CSS3 color strings.<br>
Lightweight.<br>
Reliable (Provides tests, handles all errors to avoid panic!s).<br>

Not 100% spec compliant in the name of convenience (see examples below):<br>
  * allows for extra whitespaces<br>
  * allows for floats where standard allows percentages only<br>
```

##### rendered.html

```html
<p>Easy-to-use Rust parser for CSS3 color strings.<br>
Lightweight.<br>
Reliable (Provides tests, handles all errors to avoid panic!s).<br></p>
<p>Not 100% spec compliant in the name of convenience (see examples below):<br></p>
<ul>
<li>allows for extra whitespaces<br></li>
<li>allows for floats where standard allows percentages only<br></li>
</ul>
```

##### cleaned.html

```html
<p>Easy-to-use Rust parser for CSS3 color strings.<br>
Lightweight.<br>
Reliable (Provides tests, handles all errors to avoid panic!s).<br></br></br></br></p>
<p>Not 100% spec compliant in the name of convenience (see examples below):<br></br></p>
<ul>
<li>allows for extra whitespaces<br></br></li>
<li>allows for floats where standard allows percentages only<br></br></li>
</ul>
```

**Comment by Eijebong** [comment]
This i fixed in ammonia 0.6 :)

**Comment by carols10cents** [comment]
Upgraded, deployed, and confirmed fixed!!
